tests/php/Jobs/CloneImagesOrVideosTest.php

```diff
@@ -0,0 +1,458 @@
+
+        $oldImage = ImageTest::create([
+            'filename' => 'a.jpg',
+            'taken_at' => Carbon::now()->setTimezone('Europe/Lisbon'),
+            'volume_id' => $volume->id,
+            'lng' => 1.5,
+            'lat' => 5.3,
+            'tiled' => true])->fresh();
+        ImageLabelTest::create(['image_id' => $oldImage->id]);
+        $oldImageLabel = $oldImage->labels()->first();
+
+        $request = new Request(['project' => $project, 'volume' => $volume, 'clone_file_labels' => true]);
+
+        Queue::fake();
+        $this->expectsEvents('volume.cloned');
+        with(new CloneImagesOrVideos($request))->handle();
+        Queue::assertPushed(ProcessNewVolumeFiles::class);
+
+        $copy = $project->volumes()->first();
+        $newImage = $copy->images()->first();
+        $newImageLabel = $newImage->labels()->first();
+
+        $this->assertNotNull($newImageLabel);
+        $this->assertNotNull($newImage);
+        $this->assertEquals($volume->images()->count(), $copy->images()->count());
+        $this->assertNotEquals($oldImage->id, $newImage->id);
+        $this->assertNotEquals($oldImage->uuid, $newImage->uuid);
+        $this->assertEquals($copy->id, $newImage->volume_id);
+        $this->assertNotEquals($oldImageLabel->id, $newImageLabel->id);
+        $this->assertNotEquals($oldImageLabel->image_id, $newImageLabel->image_id);
+
+        $ignore = ['id', 'volume_id', 'uuid'];
+        $this->assertEquals(
+            $oldImage->makeHidden($ignore)->toArray(),
+            $newImage->makeHidden($ignore)->toArray()
+        );
+
+        $ignore = ['id', 'image_id'];
+        $this->assertEquals(
+            $oldImageLabel->makeHidden($ignore)->toArray(),
+            $newImageLabel->makeHidden($ignore)->toArray()
+        );
+
+        // second test cloning only a subset of labels
+
+        $volume = VolumeTest::create([
+            'media_type_id' => MediaType::imageId(),
+            'created_at' => '2022-11-09 14:37:00',
+            'updated_at' => '2022-11-09 14:37:00',
+        ])->fresh(); // Use fresh() to load even the null fields.
+        // The target project.
+        $project = ProjectTest::create();
+
+        $this->beAdmin();
+        $project->addUserId($this->admin()->id, Role::adminId());
+
+        $oldImage->volume_id = $volume->id;
+        $oldImage->save();
+        // there are three labels in total
+        $l2 = ImageLabelTest::create(['image_id' => $oldImage->id]);
+        $l3 = ImageLabelTest::create(['image_id' => $oldImage->id]);
+
+        $request = new Request(['project' => $project, 'volume' => $volume, 'clone_file_labels' => true,
+            'only_file_labels' => [$l2->label_id, $l3->label_id]]);
+
+        with(new CloneImagesOrVideos($request))->handle();
+
+        $copy = $project->volumes()->first();
+        $newImage = $copy->images()->first();
+        $newImageLabel = $newImage->labels()->get();
+
+        $this->assertEquals(2, $newImageLabel->count());
```

Comment:
You should also check if the label IDs of the cloned image labels are correct. E.g.:
```php
$newImageLabels = $newImage->labels()->pluck('label_id');
$this->assertCount(2, $newImageLabels);
$this->assertContains($l2->label_id, $newImageLabels);
$this->assertContains($l3->label_id, $newImageLabels);
```